PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -1101,6 +1101,8 @@ void PATMuonProducer::embedHighLevel(pat::Muon& aMuon,
       IPTools::signedTransverseImpactParameter(tt, GlobalVector(track->px(), track->py(), track->pz()), primaryVertex);
   double d0_corr = result.second.value();
   double d0_err = primaryVertexIsValid ? result.second.error() : -1.0;
+  // Now correct the sign using information from the track
+  d0_corr = (track->dxy(primaryVertex.position()) > 0 ? 1 : -1) * fabs(d0_corr);
```

Comment:
this can be 

```suggestion
  d0_corr = std::copysign(d0_corr, track->dxy(primaryVertex.position()));
```

Reply:
actually, I'd modify on L1102 instead. There is no significant need to assign one value and then to apply this correction.